src/test/java/bio/terra/service/auth/iam/IamServiceTest.java

```diff
@@ -244,4 +252,58 @@ void testVerifyResourceTypeAdminAuthorizedFalse() throws InterruptedException {
             iamService.verifyResourceTypeAdminAuthorized(
                 TEST_USER, IamResourceType.DATASNAPSHOT, IamAction.ADMIN_READ_SUMMARY_INFORMATION));
   }
+
+  @Test
+  void testConstructFirecloudGroupName() {
+    String id = "123456";
+    assertEquals(iamService.constructFirecloudGroupName(id), String.format("%s-users", id));
+  }
+
+  @Test
+  void testConstructUniqueFirecloudGroupName() {
+    String id = "123456";
+    String actual = iamService.constructUniqueFirecloudGroupName(id);
+    // we don't know exactly what our backstop name will be as it will be suffixed by a new UUID,
+    // but we know that it will start with our more readable group name.
+    assertTrue(actual.contains(String.format("%s-users-", id)));
```

Comment:
I think hamcrest matchers are clearer here. Using a matcher will also produce a more helpful error if the assert fails.
```suggestion
    assertThat(actual, startsWith("%s-users-".formatted(id)));
```